js/index.js

```diff
@@ -8,35 +8,121 @@ const siteContent = {
+    address: "123 Way 456 Street Somewhere, USA",
+    phone: "1 (888) 888-8888",
+    email: "sales@greatidea.io"
   },
+  footer: {
+    copyright: "Copyright Great Idea! 2018"
+  }
 };
 
 // Example: Update the img src for the logo
+//let logo = document.getElementById("logo-img");
+//logo.setAttribute("src", siteContent["nav"]["img-src"]);
+
+// nav selectors
+const [nav] = document.getElementsByTagName("nav");
 let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+
+// cta selectors
+const [cta] = document.getElementsByClassName("cta");
+const [ctaH1] = cta.getElementsByTagName("h1");
+const [ctaBtn] = cta.getElementsByTagName("button");
+const ctaImg = document.getElementById("cta-img");
+
+//main content selectors
+const [maincontent] = document.getElementsByClassName("main-content");
+const myh4 = maincontent.getElementsByTagName("h4");
+const myp = maincontent.querySelectorAll("p");
+const mainImg = document.getElementById("middle-img");
+
+//contact section selectors
+const [contact] = document.getElementsByClassName("contact");
+const mycontacth4 = contact.getElementsByTagName("h4");
+const mycontactp = contact.querySelectorAll("p");
+
+//footer section selectors
+const [footer] = document.getElementsByTagName("footer");
+
+// update nav links
+Array.from(nav.children).forEach((val, idx) => {
+  val.textContent = siteContent["nav"][`nav-item-${idx + 1}`];
+});
+
+//make navigation green
+const navchange = document.querySelectorAll("a");
+
+navchange[0].style.color = "green";
+navchange[1].style.color = "green";
+navchange[2].style.color = "green";
+navchange[3].style.color = "green";
+navchange[4].style.color = "green";
+navchange[5].style.color = "green";
+
+//add to navigation
+const addBlog = document.createTextNode("Blog");
+const addHome = document.createTextNode("Home");
+
+const navadd = document.querySelector("nav");
+navadd.style.color = "green";
+
+navadd.appendChild(addBlog);
+navadd.prepend(addHome);
+
+// update img src for the logo
+logo.setAttribute("src", siteContent["nav"]["img-src"]);
+
+// update cta section
+ctaH1.textContent = siteContent["cta"]["h1"];
+ctaBtn.textContent = siteContent["cta"]["button"];
+ctaImg.src = siteContent["cta"]["img-src"];
+
+//update main content section
+myh4[0].textContent = siteContent["main-content"]["features-h4"];
+myh4[1].textContent = siteContent["main-content"]["about-h4"];
+myh4[2].textContent = siteContent["main-content"]["services-h4"];
+myh4[3].textContent = siteContent["main-content"]["product-h4"];
+myh4[4].textContent = siteContent["main-content"]["vision-h4"];
+
+myp[0].textContent = siteContent["main-content"]["features-content"];
+myp[1].textContent = siteContent["main-content"]["about-content"];
+myp[2].textContent = siteContent["main-content"]["services-content"];
+myp[3].textContent = siteContent["main-content"]["product-content"];
+myp[4].textContent = siteContent["main-content"]["vision-content"];
```

Comment:
Ahhh! More repetition. How could we reduce the amount of repeated code here?

Well, I see another opportunity for loops! We just need to make sure that we're referring to the correct `siteContent` values for each `h4` and `content` key. If we check out `siteContent`, we can see that the order in which our keys show up are the same as how they are presented on the DOM.

So, we could just grab the `h4` items in an array and map the indexes to the `myh4` nodes we gathered.

For example:
```js
// first, we grab all the `siteContent` keys into an array
const mainContentKeys = Object.keys(siteContent['main-content']);
// now we need to grab the h4 keys specifically - luckily every h4 element ends with the string 'h4'
const mainContentH4s = mainContentKeys.filter(key => key.endsWith('h4'));
// and finally, we can loop through our h4 elements and set them using the equivalent index of our `textContentH$` array
mainContentH4s.forEach((value, idx) => {
  myh4[idx].textContent = siteContent['main-content'][value];
});
```